addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animation_sampler_keyframes.py

```diff
@@ -62,12 +62,18 @@ def get_target_len(self):
         return length
 
     def __set_indexed(self, value):
+        if value is None:
+            return value
+        if "rotation" in self.target and type(value) is mathutils.Quaternion:
+            return value
         # 'value' targets don't use keyframe.array_index
         if self.target == "value":
             return value
         # Sometimes blender animations only reference a subset of components of a data target. Keyframe should always
         # contain a complete Vector/ Quaternion --> use the array_index value of the keyframe to set components in such
         # structures
+        if type(value) is mathutils.Quaternion:
+            value = tuple(value.to_euler("XYZ"))
```

Comment:
I don't understand why you convert to euler, indexes will not match anymore (if this code is really execute, see previous comment)

Reply:
The existing implementation already returns an array supporting **location**, and **scale** which are effectively XYZ eulers. In the event a Quaternion is passed, the conversion satisfies the expected simple array/euler.


Keep in mind that prior to this patch, the implementation mangled the Quaternion into a euler which caused the related bug.